source/tools/configuration.py

```diff
@@ -79,8 +80,11 @@ def uninitialize(cls):
         :return: None
         :rtype: NoneType
         """
-        node_id = SettingList.get_setting_by_code(code='node_id').value
-        if node_id is not None and cls.dir_exists(Configuration.ASD_NODE_LOCATION.format(node_id)):
+        try:
+            node_id = SettingList.get_setting_by_code(code='node_id').value
+        except ObjectNotFoundException:
+            return
```

Comment:
If the setting has been deleted already and thus this error is raised, shouldn't we still attempt to delete the `ASD_NODE_LOCATION`?

Reply:
How would we do that? The node id is necessary to determine the path